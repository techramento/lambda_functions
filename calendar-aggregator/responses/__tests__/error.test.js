const errorResponse = require('../error')
const MockErrors = require('../../__fixtures__/errors.json')
const MockInput = Object.freeze({})

describe('error responses', () => {
  it('calls the callback with an error object', () => {
    errorResponse(MockInput, (error, response) => {
      expect(error).toMatchSnapshot()
      expect(response).toMatchSnapshot()
    })(MockErrors.apiError)
  })

  it('provides a default error code when one is not provided', () => {
    errorResponse(MockInput, (error, response) => {
      expect(error).toMatchSnapshot()
      expect(response).toMatchSnapshot()
    })(MockErrors.exception)
  })

  it('does not crash if the callback itself throws', () => {
    const exception = new Error('MOCK EXCEPTION')
    const origConsole = console.error
    console.error = jest.fn()
    errorResponse(MockInput, () => {
      throw exception
    })(MockErrors.apiError)
    expect(console.error).toHaveBeenCalledWith(exception)
    console.error = origConsole
  })
})
