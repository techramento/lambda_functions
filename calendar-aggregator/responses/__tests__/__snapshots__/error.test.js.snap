// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error responses calls the callback with an error object 1`] = `
Object {
  "code": 999,
  "errors": Array [
    Object {
      "domain": "usageLimits",
      "message": "Bad Request",
      "reason": "keyInvalid",
    },
  ],
  "message": "mock error",
}
`;

exports[`error responses calls the callback with an error object 2`] = `
Object {
  "body": "{\\"details\\":\\"[{\\\\\\"domain\\\\\\":\\\\\\"usageLimits\\\\\\",\\\\\\"message\\\\\\":\\\\\\"Bad Request\\\\\\",\\\\\\"reason\\\\\\":\\\\\\"keyInvalid\\\\\\"}]\\",\\"error\\":{\\"message\\":\\"mock error\\"},\\"input\\":{},\\"message\\":\\"undefined: mock error\\"}",
  "statusCode": 999,
}
`;

exports[`error responses provides a default error code when one is not provided 1`] = `
Object {
  "message": "Mock Exception",
  "name": "ReferenceError",
}
`;

exports[`error responses provides a default error code when one is not provided 2`] = `
Object {
  "body": "{\\"error\\":{\\"message\\":\\"Mock Exception\\",\\"name\\":\\"ReferenceError\\"},\\"input\\":{},\\"message\\":\\"ReferenceError: Mock Exception\\"}",
  "statusCode": 500,
}
`;
